tests/virtual_machines_test.go

```diff
@@ -32,6 +34,26 @@ var _ = Describe("Virtual Machines", func() {
 		Expect(result.Error()).NotTo(HaveOccurred())
 	})
 
+	BeforeEach(func() {
+		// Verify that there are no VMs in the cluster
+		currentVMList := &kubevirtv1.VirtualMachineList{}
+		err := testClient.VirtClient.List(context.TODO(), currentVMList, &client.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(len(currentVMList.Items)).To(BeZero())
+
+		// Clear vmWaitConfigMap configMap data before each test
+		vmWaitConfigMap, err := testClient.KubeClient.CoreV1().ConfigMaps(names.MANAGER_NAMESPACE).Get(names.WAITING_VMS_CONFIGMAP, meta_v1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		// Clear the map in-place instead of waiting to garbage-collector
+		for entry := range vmWaitConfigMap.Data {
+			delete(vmWaitConfigMap.Data, entry)
+		}
+		vmWaitConfigMap, err = testClient.KubeClient.CoreV1().ConfigMaps(names.MANAGER_NAMESPACE).Update(vmWaitConfigMap)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(vmWaitConfigMap.Data).To(BeEmpty())
```

Comment:
same here. maybe warp in eventually because it could  take time. not sure.

Reply:
This is a double check that there is no VM in the cluster, the deletion is done in AfterEach block and it is done with eventually.

Reply:
Take a look at this [1] and [2] 
Update invokes Do which get request as a receiver,
and according to the comment in the code we will get the response synchronously, 
so Eventualy seems redundant here, please correct me if I am wrong :)

[1] https://github.com/kubernetes/client-go/blob/v11.0.0/rest/request.go#L674 
[2] https://github.com/kubernetes/client-go/blob/v11.0.0/rest/request.go#L800